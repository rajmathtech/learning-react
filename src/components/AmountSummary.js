import React from 'react';
import numeral from 'numeral';
import totalAmount from '../redux/selector/totalAmountObjects';
const AmountSummary = (props) => {
    const len = props.objects.length;
    const total = numeral(totalAmount(props.objects)/100).format('$0,0.00');
    return (<div>
    <h1> You are viewing { len > 1 ? ` ${len} objects of total ${total}` : ` ${len} object of total ${total}` }</h1>
</div>);
    }
export default AmountSummary;