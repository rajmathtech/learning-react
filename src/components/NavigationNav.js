import React from 'react';
import { NavLink } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../redux/actions/auth';
export const NavigationNav = ({startLogout}) => (

<nav className="navigation">
    <ul className="navigation__list">
        <li className="navigation__item">
            <NavLink to="/objects" activeClassName="active-link" className="navigation__link">
                Home
            </NavLink>
        </li>
        <li className="navigation__item">
            <NavLink to="/create" activeClassName="active-link" className="navigation__link">
                Create
            </NavLink>
        </li>
        <li className="navigation__item">
            <button onClick = {startLogout}> Logout </button>
        </li>
    </ul>
</nav>

);

const mapDispatchToProps = (dispatch) =>{
    return {
        startLogout: () => dispatch(startLogout())
    }
}
export default connect(undefined, mapDispatchToProps)(NavigationNav);