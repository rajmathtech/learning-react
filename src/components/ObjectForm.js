import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";
export default class ObjectForm extends React.Component {
  constructor(props) {
    super(props);
    const { title, details, amount, createdAt } = this.props.object;
    this.state = {
      title: title ? title : "",
      details: details ? details : "",
      createdAt: createdAt ? moment(createdAt) : moment(),
      amount: amount ? (amount / 100).toString() : "",
      error: "",
      focused: false
    };
  }
  onDateChange = createdAt => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ focused }));
  };
  render() {
    return (
      <div>
        <h1>from Object form </h1>
        <h2>{this.state.error}</h2>
        <form onSubmit={this.onSubmitForm}>
          <input
            type="text"
            placeholder="title"
            value={this.state.title}
            onChange={this.onTitleChange}
          />

          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.focused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />

          <textarea
            name="details"
            rows="10"
            cols="50"
            placeholder="details"
            value={this.state.details}
            onChange={this.onDetailsChange}
          />
          <input
            type="text"
            placeholder="amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <button type="submit"> Add </button>
        </form>
      </div>
    );
  }
  onSubmitForm = e => {
    e.preventDefault();

    if (!this.state.amount || !this.state.title) {
      this.setState(() => ({ error: "Please enter title and amount!" }));
    } else {
      this.props.onSubmitForm({
        ...this.state,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf()
      });
    }
  };
  onTitleChange = e => {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };
  onDetailsChange = e => {
    const details = e.target.value;
    this.setState(() => ({ details }));
  };
  onAmountChange = e => {
    const amount = e.target.value;
    if (amount.match(/^\d*(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };
}
