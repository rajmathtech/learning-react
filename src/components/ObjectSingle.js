import React from 'react';
import {Link} from 'react-router-dom';
const ObjectSingle = (props) => (
    <div className="singleObject"> 
        <Link to={`display/${props.object.id}`} className="objectLink" >
            {props.object.title.toUpperCase()}
        </Link>
    </div>
);
export default ObjectSingle;