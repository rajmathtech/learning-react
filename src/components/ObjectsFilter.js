import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setText, setSortBy, setStartDate, setEndDate } from '../redux/actions/filter';

export class ObjectsFilter extends React.Component {
  state = {
    focused: null
  };
  onChangeText = (e) => {
    const text = e.target.value;
    this.props.setText(text);
    }
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (focused) => {
    this.setState(() => ({ focused }));
  }
  render() {
    return (
      <div>
      <input type="text" value={this.props.filter.text} 
      onChange={this.onChangeText} />
        <select
          value={this.props.filter.sortBy}
          onChange={(e) => {
            if (e.target.value === 'date') {
              this.props.setSortBy('date');
            } else if (e.target.value === 'amount') {
              this.props.setSortBy('amount');
            }
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filter.startDate}
          endDate={this.props.filter .endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.focused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  }
};

const mapStateToProps = (state) => ({ filter: state.filter });
const mapDispatchToProps = (dispatch) => ({
  setText: (text) => dispatch(setText(text)),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setSortBy: (by) => dispatch(setSortBy(by))

});
export default connect(mapStateToProps, mapDispatchToProps)(ObjectsFilter);
