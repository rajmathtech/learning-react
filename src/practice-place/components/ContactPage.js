import React from 'react';
const ContactPage = () => (<div>
    <h1> welcome to contact page</h1>
    <p>this is the details of the contact page!</p>
</div>);
export default ContactPage;