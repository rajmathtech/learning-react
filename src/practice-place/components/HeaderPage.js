import React from 'react';
import {NavLink} from 'react-router-dom';
const HeaderPage = () => (<div>
    <NavLink to="/" exact activeClassName="active-link"> home </NavLink>
    <NavLink to="/contact" activeClassName="active-link"> contact </NavLink>
    <NavLink to="/users" activeClassName="active-link"> users </NavLink>
</div>);
export default HeaderPage;