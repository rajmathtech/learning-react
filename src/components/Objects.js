import React from "react";
import ObjectSingle from "./ObjectSingle";
import { connect } from "react-redux";
import filteredObjects from "../redux/selector/filteredObjects";
import ObjectsFilter from "./ObjectsFilter";
export const Objects = props => (
  <div className="objects-container">
    <ObjectsFilter />
    {props.objects.length === 0 ? (
      <p> no object found! </p>
    ) : (
      props.objects.map(object => (
        <ObjectSingle key={object.id} object={object} />
      ))
    )}
  </div>
);

const mapStateToProps = ({ objects, filter }) => {
  // console.log(objects);
  // console.log(filter);
  // console.log(filteredObjects(objects, filter));
  // console.log('-----');

  return {
    objects: filteredObjects(objects, filter),
    filter
  };
};
export default connect(mapStateToProps)(Objects);
