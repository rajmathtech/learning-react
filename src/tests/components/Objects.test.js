import React from 'react';
import {Objects} from '../../components/Objects';
import objects from '../fixtures/objects';
import {shallow} from 'enzyme';
test('should render with no objects passed', () => {
    const wrapper = shallow(<Objects objects={[]}/>);
    expect(wrapper).toMatchSnapshot();  
});
test('should render with objects passed', () => {
    const wrapper = shallow(<Objects objects={objects}/>);
    expect(wrapper).toMatchSnapshot();  
});