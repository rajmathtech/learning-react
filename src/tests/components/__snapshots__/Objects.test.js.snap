// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with no objects passed 1`] = `
<div
  className="objects-container"
>
  <AmountSummary
    objects={Array []}
  />
  <Connect(ObjectsFilter) />
  <p>
     no object found! 
  </p>
</div>
`;

exports[`should render with objects passed 1`] = `
<div
  className="objects-container"
>
  <AmountSummary
    objects={
      Array [
        Object {
          "amount": 400,
          "createdAt": -345600000,
          "details": " details of object one",
          "id": "1",
          "title": "object one",
        },
        Object {
          "amount": 300,
          "createdAt": 0,
          "details": " details of object two",
          "id": "2",
          "title": "object two",
        },
        Object {
          "amount": 500,
          "createdAt": 345600000,
          "details": " details of object three",
          "id": "3",
          "title": "object three",
        },
      ]
    }
  />
  <Connect(ObjectsFilter) />
  <ObjectSingle
    key="1"
    object={
      Object {
        "amount": 400,
        "createdAt": -345600000,
        "details": " details of object one",
        "id": "1",
        "title": "object one",
      }
    }
  />
  <ObjectSingle
    key="2"
    object={
      Object {
        "amount": 300,
        "createdAt": 0,
        "details": " details of object two",
        "id": "2",
        "title": "object two",
      }
    }
  />
  <ObjectSingle
    key="3"
    object={
      Object {
        "amount": 500,
        "createdAt": 345600000,
        "details": " details of object three",
        "id": "3",
        "title": "object three",
      }
    }
  />
</div>
`;
